### PR TITLE
Fixes ros-noetic-arch/ros-noetic-moveit-setup-assistant#3

### PR DESCRIPTION
**Pull request by AM4283**
This should fix the build issue, package installed correctly after applying this patch

Patches #3

### CODE REVIEW

**Review by acxz — CHANGES_REQUESTED** [review]
Sweet, glad you got to the bottom of this!

Can you update the pkgrel and the .SRCINFO as well? See: https://wiki.archlinux.org/title/PKGBUILD#pkgrel and https://wiki.archlinux.org/title/.SRCINFO

Also can you make a pull request to upstream and then change the source to point to the PR instead. I.e. something like this commit: https://github.com/ros-noetic-arch/ros-noetic-octomap/commit/4b75e6dc63641964d4bc44c89c199f8d4af69656

Can you also change the commit message to be like so:
`git commit -m "upgpkg: ros-noetic-moveit-setup-assistant 1.1.8-4" -m "apply patch for proper bool type"`

**Comment by AM4283** [comment]
Ok should be done now!

File: PKGBUILD [hunk 1]
```diff
@@ -30,8 +30,13 @@ depends=(${ros_depends[@]}
   ompl)
 
 _dir="moveit-${pkgver}/moveit_setup_assistant"
-source=("${pkgname}-${pkgver}.tar.gz"::"https://github.com/ros-planning/moveit/archive/${pkgver}.tar.gz")
-sha256sums=('2a88440169593037c4adbf14896c30def63f8b3af85f1239e8ef94ee62b0b969')
+source=("${pkgname}-${pkgver}.tar.gz"::"https://github.com/ros-planning/moveit/archive/${pkgver}.tar.gz" "fixbuild.patch"::"https://patch-diff.githubusercontent.com/raw/ros-planning/moveit/pull/3175.patch")
```
Comment (acxz) [inline PKGBUILD]: just a nit, can you put the patch source on the next line to keep it consistant

File: .SRCINFO [hunk 2]
```diff
@@ -29,6 +29,8 @@ pkgbase = ros-noetic-moveit-setup-assistant
 	depends = log4cxx
 	depends = ompl
 	source = ros-noetic-moveit-setup-assistant-1.1.8.tar.gz::https://github.com/ros-planning/moveit/archive/1.1.8.tar.gz
+	source = build.patch::https://patch-diff.githubusercontent.com/raw/ros-planning/moveit/pull/3175.patch
```
Comment (acxz) [inline .SRCINFO]: Looks like you need to run `makepkg --printsrcinfo > .SRCINFO` again, the source (build.patch) name is different from the one in PKGBUILD (fixbuild.patch)

File: fixbuild.patch [hunk 3]
```diff
@@ -0,0 +1,12 @@
+diff --unified --recursive --text moveit-1.1.8/moveit_setup_assistant/src/tools/moveit_config_data.cpp moveit-1.1.8.new/moveit_setup_assistant/src/tools/moveit_config_data.cpp
```
Comment (acxz) [inline fixbuild.patch]: You can delete this, since the patch source is obtained online.

**Review by acxz — CHANGES_REQUESTED** [review]

**Comment by acxz** [comment]
The AUR will only accept one commit per update, can you squash your commits into one?

**Comment by acxz** [comment]
@AM4283 can you follow up on the comments?

**Comment by AM4283** [comment]
> @AM4283 can you follow up on the comments?

Sorry, haven't been active on my personal GitHub in a bit. Hopefully this is good? This is the first time I've squashed commits like this so I'm not sure if I did it exactly right.

**Comment by acxz** [comment]
I'm afraid you seem to have added commits instead of squashed them. I recommend using the `git rebase -i` command.

**Comment by acxz** [comment]
If you want we can get on call to work this out together, or you can give me write access to your fork and I can fix it myself.

**Comment by acxz** [comment]
closed with [104f592](https://github.com/ros-noetic-arch/ros-noetic-moveit-setup-assistant/commit/104f59217b0db7d819c56594f6dd71f333962625)

Sorry @AM4283 thought it would easier if I just pulled in the commit myself. But we got your commit in! Thanks for the contribution!

**Pull request closed without merging**